pkgs/development/compilers/dotnet/build-dotnet.nix

```diff
@@ -46,30 +50,57 @@ let
     nugetDeps = packages;
   } else null;
 
+  _icu = if stdenv.isDarwin then darwin.ICU else icu;
```

Comment:
Does this work the nixpngs icu package on Darwin, or does it have to use darwin.ICU? darwin.ICU is pretty old, and I’d like to deprecate and remove it eventually (since almost nothing uses it). If .NET needs it for ABI reasons, I can look at what it would take to fix it like I did libiconv.

Reply:
It doesn't 'just work' with `icu` (same error).  I'll have to git into how ICU loads data to know why exactly.

Reply:
I can get it to load `icu` if I make a shim package with a `libicucore.A.dylib` symlink to `libicuuc.dylib`.

However, it fails like:

```
dyld[96379]: dlsym(0x82baaca0, "u_charsToUChars")
dyld[96379]:      dlsym("u_charsToUChars") => NULL
dyld[96379]: dlerror()dyld[96379]:  => 'dlsym(0x82baaca0, u_charsToUChars): symbol not found'
Cannot get symbol u_charsToUChars from libicuuc
Error: dlsym(0x82baaca0, u_charsToUChars): symbol not found
/nix/store/zf8hff9bnlfix7l7gjsnv3jc22lxzzxx-stdenv-darwin/setup: line 1591: 96379 Abort trap: 6           DYLD_PRINT_LIBRARIES=1 DYLD_PRINT_APIS=1 $out/dotnet
```

The symbols in `libicuuc.dylib` have a version suffix, so `_u_charsToUChars_73` in this case, and the pal shim doesn't append that on darwin.

If we renamed the symbols, we'd probably also have to deal with this assumption that it makes:

```
    // in OSX all ICU APIs exist in the same library libicucore.A.dylib
    libicui18n = libicuuc;
```